File: 16.py
class HashTable:
    def __init__(self, size=10):
        self.size = size
        self.table = [[] for _ in range(size)]

    def hash(self, key):
        return key % self.size

    def insert(self, key):
        idx = self.hash(key)
        if key not in self.table[idx]:
            self.table[idx].append(key)

    def search(self, key):
        idx = self.hash(key)
        return key in self.table[idx]

    def remove(self, key):
        idx = self.hash(key)
        if key in self.table[idx]:
            self.table[idx].remove(key)

File: 16.js
class HashTable {
    constructor(size = 10) {
        this.size = size;
        this.table = Array.from({ length: size }, () => []);
    }

    hash(key) {
        return key % this.size;
    }

    insert(key) {
        const idx = this.hash(key);
        if (!this.table[idx].includes(key)) {
            this.table[idx].push(key);
        }
    }

    search(key) {
        return this.table[this.hash(key)].includes(key);
    }

    remove(key) {
        const idx = this.hash(key);
        this.table[idx] = this.table[idx].filter(x => x !== key);
    }
}
